src/main/java/LinkedBinarySearchTree.java

```diff
@@ -0,0 +1,492 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class LinkedBinarySearchTree<T extends Comparable<? super T>> implements BinarySearchTree<T> {
+
+    private int size;
+    private Node<T> root;
+
+    public LinkedBinarySearchTree() {
+        root = null;
+        size = 0;
+    }
+
+    public LinkedBinarySearchTree(T element) {
+        root = new Node<>(element);
+        size = 0;
+    }
+
+    @Override
+    public void add(T element) {
+        if (isEmpty()) {
+            root = new Node<>(element);
+        } else {
+            add(element, root);
+        }
+        size++;
+    }
+
+    /**
+     * Helper add method for enabling recursive add.
+     *
+     * @param element
+     * @param current
+     */
+    private void add(T element, Node<T> current) {
+        if (current.getData().compareTo(element) > 0) {
+            if (current.getLeft() == null) {
+                current.setLeft(new Node<>(element));
+            } else {
+                add(element, current.getLeft());
+            }
+        } else {
+            if (current.getRight() == null) {
+                current.setRight(new Node<>(element));
+            } else {
+                add(element, current.getRight());
+            }
+        }
+    }
```

Comment:
This is quite the mouthful. You can simplify a bit by returning a node, and make the function look more like a basic recursion example.

```java
private Node<T> add(T t, Node<T> node) {
    if (node == null) { return new Node<>(t); }
    else if (t < node.value) { node.left = add(t, node.left); }
    else { node.right = add(t, node.right); }
    return node;
}
```
`compareTo` and other methods left out for brevity. You'd also need to change the public `add` method slightly;

```java
public void add(T t) {
    root = add(t, root);
}
```